### DISCUSSION TITLE
failed to start [/usr/sbin/iptables-restore --wait 30 --wait-interval 100000]: exit status 2: unknown

### DISCUSSION THREAD

**Discussion by johnwmail** [wollomatic/socket-proxy#37] [Q&A] [answered]
docker-compose:
![image](https://github.com/user-attachments/assets/52387a19-0444-4bb4-ac22-528ef561cce6)

error messages:
![image](https://github.com/user-attachments/assets/9813a518-1370-42f1-9b80-15d99da66453)

/etc/docker/daemon.json:
![image](https://github.com/user-attachments/assets/ef9444cc-2e3c-485f-b0ed-0b49877b02fc)

Any workaround how to solve it?   thanks.

**Comment by wollomatic** [comment]
I'm not familiar with sysbox-runc. Socket-proxy is just a SCRATCH image, so there is no `/usr/sbin/iptables-restore` binary in the container image. There is also no `glibc` or `musl` in the image that could be needed to run iptables-restore.

Maybe it's worth a try to create a container image based on a distroless or a minimal linux image, like `gcr.io/distroless/base-debian12:nonroot`, `alpine:3.20`, or even Ubuntu.

Sample Dockerfile:
```
FROM wollomatic/socket-proxy:1 AS proxy
FROM gcr.io/distroless/base-debian12:nonroot AS base
VOLUME /var/run/docker.sock
EXPOSE 2375
ENTRYPOINT ["/socket-proxy"]
COPY --from=proxy ./healthcheck ./socket-proxy /
```

**Comment by johnwmail** [comment, marked as answer]
Thank for your sharing, thanks.

I noticed, it works if comment out "#internal: true" (it also work, if external: false), thanks
![image](https://github.com/user-attachments/assets/c67553b1-ee7b-4953-a475-912315b8f2e2)
